src/lib.ts

```diff
@@ -575,3 +577,31 @@ export function withLocation<T>(location: Location, action: () => T): T {
     setLocation(start);
   }
 }
+
+export function useableCinch(): number {
+  if (!have($item`Cincho de Mayo`)) return 0;
+  const cinchRestored = Array(100)
+    .fill(0)
+    .map((_, i) => clamp(50 - 5 * i, 5, 30));
+  const cinchRestsUsed = get("_cinchoRests", 0);
+  const freeRestsLeft = Math.max(0, totalFreeRests() - get("timesRested"));
+  return (
+    100 -
+    get("_cinchUsed", 0) +
+    sumNumbers(cinchRestored.slice(cinchRestsUsed, cinchRestsUsed + freeRestsLeft))
+  );
+}
+
+export function freeRest(): boolean {
+  if (get("timesRested") >= totalFreeRests()) return false;
+
+  if (get("chateauAvailable")) {
+    visitUrl("place.php?whichplace=chateau&action=chateau_restlabelfree");
```

Comment:
it might be good to have some logic here that evaluates if it is profitable to the user to swap to the free rest vs the adv ceiling fan in the chateau. i do not think this is needed for first implementation but maybe a comment here to note that it would be a nice-to-have would be good.

Reply:
probably just anyone who owns cincho + a way to eke value from NC forcers should switch. the bigger issue is the Guy who gets mad we don't reinstall their antlers. I'll work on that later